pkg/webhooks/validation/server/server_test.go

```diff
@@ -223,6 +251,52 @@ func TestAdmitPilot(t *testing.T) {
 			}
 		})
 	}
+
+	smeV1 := makeServiceMeshExtensionConfig(t, "v1")
+	smeV1alpha1 := makeServiceMeshExtensionConfig(t, "v1alpha1")
+	deprecatedWarning := "ServiceMeshExtension is deprecated. Please use WasmPlugin instead."
+
+	maistraCases := []struct {
+		name    string
+		admit   admitFunc
+		in      *kube.AdmissionRequest
+		allowed bool
+		warning string
+	}{
+		{
+			name:  "v1 ServiceMeshExtension deprecation warning",
+			admit: wh.validate,
+			in: &kube.AdmissionRequest{
+				Kind:      kubeApisMeta.GroupVersionKind{Kind: "ServiceMeshExtension"},
+				Object:    runtime.RawExtension{Raw: smeV1},
+				Operation: kube.Create,
+			},
+			allowed: true,
+			warning: deprecatedWarning,
+		},
+		{
+			name:  "v1 ServiceMeshExtension deprecation warning",
+			admit: wh.validate,
+			in: &kube.AdmissionRequest{
+				Kind:      kubeApisMeta.GroupVersionKind{Kind: "ServiceMeshExtension"},
+				Object:    runtime.RawExtension{Raw: smeV1alpha1},
+				Operation: kube.Create,
+			},
+			allowed: true,
+			warning: deprecatedWarning,
+		},
+	}
+
+	for i, c := range maistraCases {
+		t.Run(fmt.Sprintf("[MAISTRA][%d] %s", i, c.name), func(t *testing.T) {
+			got := c.admit(c.in)
+			if got.Allowed != c.allowed {
+				t.Fatalf("got %v want %v", got.Allowed, c.allowed)
+			} else if got.Warnings[0] != c.warning {
+				t.Fatalf("did not get warning \"%s\"", c.warning)
+			}
```

Comment:
This `if-else` statement is absolutely correct, but since `else` is not important for the correctness of the test, couldn't it be simplified to two separate `if` statements? At the first glance it was confusing for me and I had to reflect if it matters or not. It's not important, so it's up to you.

Reply:
I have tried to improve it. The correct behavior is `allowed == true` as well as getting a warning message.

Reply:
I meant two subsequent `if` statements:
```
if got.Allowed != true {
  t.Fatal(...)
}
if got.Warnings[0] != deprecatedWarning {
  t.Fatal(...)
}
```
Having a single assertion requires manual checking what's wrong once the test fails.

Reply:
Done